pkg/new-ui/v1beta1/backend.go

```diff
@@ -421,3 +431,92 @@ func (k *KatibUIHandler) FetchTrial(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+// FetchTrialLogs fetches logs for a trial in specific namespace.
+func (k *KatibUIHandler) FetchTrialLogs(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Requesting logs")
+
+	trialName := r.URL.Query()["trialName"][0]
+	namespace := r.URL.Query()["namespace"][0]
+	log.Printf("Requesting logs")
+
+	logs, err := getTrialLogs(k, trialName, namespace)
+	if err != nil {
+		log.Printf("GetLogs failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	response, err := json.Marshal(logs)
+	if err != nil {
+		log.Printf("Marshal logs failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if _, err = w.Write(response); err != nil {
+		log.Printf("Write logs failed: %v", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+// GetTrialLogs returns logs of a master Pod for the given job name and namespace
+func getTrialLogs(k *KatibUIHandler, trialName string, namespace string) (string, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return "", err
+	}
+
+	clientset, err := corev1.NewForConfig(cfg)
```

Comment:
Can we use [the Controller Runtime client from the `KatibUIHandler`](https://github.com/kubeflow/katib/blob/4d979d9199c0c6470aec5573c4611926b045510c/pkg/util/v1beta1/katibclient/katib_client.go#L60-L81) to get the Pod logs ?

Reply:
From my understanding based on this issue https://github.com/kubernetes-sigs/controller-runtime/issues/452, it is not possible to obtain logs using the controller runtime client, we need a Kubernetes client. If I'm missing something, please let me know.

Reply:
Thanks @tenzen-y, added the comment.